tests/server.py

```diff
@@ -101,7 +102,9 @@ def process(self) -> None:
                 path = uri.path
 
                 if request_subscribers.get(path):
-                    request_subscribers[path].set_result(request)
+                    loop.call_soon_threadsafe(
```

Comment:
TIL! we call it from another thread into the testing main thread. It was racing and test_frame_wait_for_nav_should_fail_when_frame_detaches was failing then.